packages/tldraw/src/state/data/filesystem.ts

```diff
@@ -48,12 +48,15 @@ export async function saveToFileSystem(
     const hasPermissions = await checkPermissions(fileHandle)
     if (!hasPermissions) return null
   }
-
+  let filename = ''
```

Comment:
```suggestion
  let filename = file.name
```

You probably want to initialize this in case `supported` is `true`.